libraries/charting.php

```
@@ -91,15 +81,28 @@ function exportHighchart(
  *
  * @throws \Exception on invalid format, command execution failure, or non zero exit status
  */
-function getScreenFromChromium($file, $width, $height, $format){
+function getScreenFromChromium($html, $width, $height, $format){
+
+    // Chromium requires the file to have a .html extension
+    // cant use datauri as it will not execute embdeeded javascript
+    $tmpFile = tempnam(sys_get_temp_dir(), 'xdmod-chromiumHtml-');
+    $tmpHtmlFile = $tmpFile . '.html';
+    if ($tmpFile === false || rename($tmpFile, $tmpHtmlFile) === false) {
+        @unlink($tmpFile);
```

what happens if you try to `unlink(false)`

---

nothing cause `@` hides the failure

---

```
Warning: unlink(): No such file or directory
```